### DISCUSSION TITLE
可否支持DLNA？

### DISCUSSION THREAD

**Discussion by StrollStars** [hooke007/mpv_PlayKit#79] [mpv 通用]
请问可否支持DLNA，以方便在局域网内播放基于DLNA的媒体？

**Comment by hooke007** [comment]
https://github.com/chachmu/mpvDLNA
感觉更应该交给前端去实现这个功能

**Reply by StrollStars** [reply]
感觉那个实现好复杂

**Reply by zhongfly** [reply]
> 感觉那个实现好复杂

并不复杂，就是复制一下，安装一下而已。https://github.com/chachmu/mpvDLNA#installation-instructions

**Comment by zhongfly** [comment]
而且这里也不是mpv开发，想要新功能应该去开发的仓库提
https://github.com/mpv-player/mpv/issues

**Reply by StrollStars** [reply]
请问下，上层楼给的链接是第三方支持，并且需要安装Python对吧？

**Reply by zhongfly** [reply]
> 请问下，上层楼给的链接是第三方支持，并且需要安装Python对吧？

是，安装都是双击、下一步那种简单事，没有想的那么难的。

mpv的开发者认为没必要支持dlna，因为视频其实还是通过http传输，用户自己去找到实际的视频流地址就行，这样的地址是mpv已经支持的。(大意,https://github.com/mpv-player/mpv/issues/2479#issuecomment-156605123)

**Reply by zhongfly** [reply]
> 请问下，上层楼给的链接是第三方支持，并且需要安装Python对吧？

关于安装python，我只有一个小建议，别安装到系统盘就行。

**Reply by hooke007** [reply]
python 你可以直接用lazy集成的embed python，添加到环境变量就和安装版的python差不多了，embed好像没有内置pip可能需要再手动装一下

**Reply by StrollStars** [reply]
@zhongfly 我一般是提取的绿色版使用，但偶尔会有兼容性问题。
@hooke007 原来pip还能手动安装，多谢告知。

**Reply by hooke007** [reply]
embed python相比安装版是少了一些预安装的库，，兼容性上不会有区别

**Reply by StrollStars** [reply]
@hooke007 多谢告知

**Reply by StrollStars** [reply]
@hooke007 lazy集成的Python太新，Win7跑不了，我下载了官方的embed python，虽然能装pip，但是跑不起来，好像和缺少runpy.py有关。embed python可以用pip管理模块？

**Reply by hooke007** [reply]
我的开发环境一直都是embed python
![Snipaste_2022-01-03_21-04-57](https://user-images.githubusercontent.com/41094733/147933816-3f831300-bc52-4ca7-83fc-e2c149b3803a.jpg)

**Reply by StrollStars** [reply]
@hooke007 我遇到了[win10下，" No module named 'pip' "如何解决？](https://segmentfault.com/q/1010000016236939)同样的问题，现在可用了。顺带问下，用`exit()`命令无法退出Python，是缺少什么包吗？

**Reply by hooke007** [reply]
在完整配置环境变量后，
我这里什么问题都没有。。。

**Reply by StrollStars** [reply]
@hooke007 可能是不同的嵌入式压缩包中，文件的内容不同吧，感觉我用这个版本还不太完善。另外想问下，你的mpv.net_CM支持mpvDLNA吗？我按安装说明操作没效果，是不是应该把mpv.net_CM自带的Python相关文件（不适用Windows7）都删除掉？我自己部署的嵌入版已添加环境变量。

**Reply by hooke007** [reply]
不清楚 我没有用过这个脚本，，我也不使用dlna/upnp
